src/functions.php

```diff
@@ -423,13 +423,14 @@ function upload($source, $destination, array $config = [])
     $host = Context::get()->getHost();
     $source = parse($source);
     $destination = parse($destination);
+    $config['options'] = is_array($config['options']) ? $config['options'] : [];
 
     if ($host instanceof Localhost) {
         $rsync->call($host->getHostname(), $source, $destination, $config);
     } else {
         $sshArguments = $host->getSshArguments()->getCliArguments();
         if (empty($sshArguments) === false) {
-            $config['options'] = ["-e 'ssh $sshArguments'"];
```

Comment:
Alternatively add the array check here:
if (!is_array($config['options'])) {
    $config['options'] = [];
}